data/functions/urls.sql

```diff
@@ -29,3 +29,40 @@ begin
     end;
 end
 $$ language plpgsql immutable;
+
+create or replace function report_fec_url(image_number text, file_number integer) returns text as $$
+begin
+
+    return case
+        when file_number < 1 then null
+        when image_number is not null and not is_electronic(image_number) then format(
+            'http://docquery.fec.gov/paper/posted/%1$s.fec',
+            file_number
+        )
+        when image_number is not null and is_electronic(image_number) then format(
+            'http://docquery.fec.gov/dcdev/posted/%1$s.fec',
+            file_number
+        )
+    end;
+end
+$$ language plpgsql immutable;
+
+create or replace function means_filed(image_number text) returns text as $$
+begin
+    return case
+        when is_electronic(image_number) then 'e-file'
+        else 'paper'
+    end;
+end
+$$ language plpgsql immutable;
+
+
+create or replace function is_electronic(image_number text) returns boolean as $$
+begin
+    return case
+        when char_length(image_number) = 18 and substring(image_number from 9 for 1) = '9' then true
+        when char_length(image_number) = 11 and substring(image_number from 3 for 1) = '9' then true
```

Comment:
Thanks!  We should definitely document them here, I could see them getting lost and forgotten easily.